Modules/posixmodule.c

```diff
@@ -18168,6 +18545,25 @@ posixmodule_exec(PyObject *m)
     }
 #endif
 
+#ifdef HAVE_STATX
+    if (statx == NULL) {
+        PyObject* dct = PyModule_GetDict(m);
+        if (dct == NULL) {
+            return -1;
+        }
+        if (PyDict_PopString(dct, "statx", NULL) < 0) {
+            return -1;
+        }
+    }
+    else {
+        pystatx_result_spec.name = "os.statx_result";
```

Comment:
Why not using `os.statx_result` name directly in pystatx_result_spec? (line 3505)

Reply:
I don't know.  I was just following the pattern of other types created in `posixmodule_exec`, most of which assign their name here.  The tests pass using `os.statx_result` directly in the spec, so I'll make that change.

Reply:
I wrote https://github.com/python/cpython/pull/140168 to set type names earlier in posixmodule.c.